app/app.go

```diff
@@ -535,7 +539,14 @@ func NewStrideApp(
 		runtime.NewKVStoreService(keys[feegrant.StoreKey]),
 		app.AccountKeeper,
 	)
-	app.UpgradeKeeper = upgradekeeper.NewKeeper(skipUpgradeHeights,
+
+	skipUpgradeHeights := map[int64]bool{}
+	for _, h := range cast.ToIntSlice(appOpts.Get(server.FlagUnsafeSkipUpgrades)) {
+		skipUpgradeHeights[int64(h)] = true
+	}
+	homePath := cast.ToString(appOpts.Get(flags.FlagHome))
```

Comment:
i don't really like it here. maybe better to do it outside and pass homePath into here like before?

Reply:
plus since wasmvm v2 that won't work because tests that use the same homedir will deadlock, that's why i added the MkdirTemp in the tests setup